src/v2/stack_aggregator.py

```diff
@@ -316,15 +303,30 @@ def _get_denormalized_package_details(self) -> List[PackageDetails]:
             package_details.append(package_detail)
         return package_details
 
-    def _get_unknown_packages(self) -> List[Package]:
+    def get_unknown_packages(self, include_transitive=False) -> List[Package]:
         """Get list of unknown packages from the normalized_package_details."""
-        all_dependencies = set(self._normalized_packages.all_dependencies)
+        if include_transitive:
+            all_dependencies = set(self._normalized_packages.all_dependencies)
+        else:
+            all_dependencies = set(self._normalized_packages.direct_dependencies)
         analyzed_dependencies = set(self._normalized_package_details.keys())
         unknown_dependencies = list()
         for pkg in all_dependencies.difference(analyzed_dependencies):
             unknown_dependencies.append(pkg)
         return unknown_dependencies
 
+    def initiate_unknown_package_ingestion(self):
```

Comment:
I am not sure if this functionality belongs in this class.

Reply:
it can be moved outside.